Meta/http-test-server.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3
+import http.client
+import http.server
+import json
+import os
+import signal
+import socketserver
+import subprocess
+import argparse
+import sys
+import time
+from urllib.parse import urlparse
+
+# In-memory store for echo responses
+echo_store = {}
+
+
+class TestHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, directory=self.static_directory, **kwargs)
+
+    def do_GET(self):
+        if self.path == "/":
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            self.wfile.write(self.get_root_html().encode("utf-8"))
+        elif self.path == "/shutdown":
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            self.wfile.write(b"Goodbye")
+            self.server.server_close()
+            print("Goodbye")
+            sys.exit(0)
+        elif self.path == "/ping":
+            self.send_response(200)
+            self.send_header("Content-Type", "text/plain")
+            self.end_headers()
+            self.wfile.write(b"pong")
+        elif self.path.startswith("/echo/"):
+            self.handle_echo()
+        elif self.path.startswith("/static/"):
+            # Remove '/static/' prefix and use built-in method
+            self.path = self.path[7:]
+            return super().do_GET()
+        else:
+            self.send_error(404, "Not Found")
+
+    def do_POST(self):
+        if self.path == "/create":
```

Comment:
Since this is pretty dynamic already, would it be possible to setup the methods and paths without requiring a correlation ID? I think it would be more ergonomic from the client side instead of tracking IDs.

e.g.:
Test code sets up with this:
`POST /create { "method": "GET", "path": "/http-code-test", "status": "400", "writes": "hello world" }`

Test makes a request to: http://localhost:8000/http-code-test

Reply:
Ya that's certainly doable and I like that there's no ID. 

Since the server is intended to be running during the entire duration of the test suite how would you expect to  deal with "collisions" between tests that tried to use the same method/path? 

E.g. test 1 creates `GET /foo { status: 404 }` and test 2 creates `GET /foo { status: 307 }`

Reply:
It can probably just return a 400: Bad Request if one already exists. The client code can then ensure a success code is received before proceeding. Should make it obvious quickly if anyone does create colliding names.